backend/vkt/src/main/java/fi/oph/vkt/service/PaymentService.java

```diff
@@ -165,38 +164,31 @@ public String createPaymentForEnrollment(final Long enrollmentId, final Person p
     final List<Item> itemList = getItems(enrollment);
     final Customer customer = Customer
       .builder()
-      .email(enrollment.getEmail())
-      .phone(enrollment.getPhoneNumber())
-      .firstName(person.getFirstName())
-      .lastName(person.getLastName())
+      .email(getCustomerField(enrollment.getEmail(), Customer.EMAIL_MAX_LENGTH))
+      .phone(getCustomerField(enrollment.getPhoneNumber(), Customer.PHONE_MAX_LENGTH))
+      .firstName(getCustomerField(person.getFirstName(), Customer.FIRST_NAME_MAX_LENGTH))
+      .lastName(getCustomerField(person.getLastName(), Customer.LAST_NAME_MAX_LENGTH))
       .build();
 
-    final int total = getTotal(itemList);
+    final int amount = getTotalAmount(itemList);
+
     final Payment payment = new Payment();
-    payment.setAmount(total);
     payment.setEnrollment(enrollment);
+    payment.setAmount(amount);
     paymentRepository.saveAndFlush(payment);
 
-    final PaytrailResponseDTO response = paytrailService.createPayment(itemList, payment.getId(), customer, total);
-
-    final String transactionId = response.getTransactionId();
-    final String reference = response.getReference();
-    final String paymentUrl = response.getHref();
+    final PaytrailResponseDTO response = paymentProvider.createPayment(itemList, payment.getId(), customer, amount);
 
+    payment.setTransactionId(response.getTransactionId());
+    payment.setReference(response.getReference());
+    payment.setPaymentUrl(response.getHref());
     payment.setPaymentStatus(PaymentStatus.NEW);
-    payment.setReference(reference);
-    payment.setPaymentUrl(paymentUrl);
-    payment.setTransactionId(transactionId);
     paymentRepository.saveAndFlush(payment);
 
-    updateEnrollmentStatus(enrollment, PaymentStatus.NEW);
```

Comment:
Tämä ei tainnut olla oleellinen?

Reply:
Hmm, voikohan käydä niin, että jos enrollment on cancel tilassa ja sitä lähdetään maksamaan uudestaan niin se ei enää muutu expecting_payment tilaan. 🤔

Reply:
Aivan, tuohon tilanteeseen tä tais ollakin. Mietin että tää status asetetaan expecting payment tilaan nyt kahdesti kun alussa enrollmentia luodessa ja seuraavaksi heti perään maksulinkkiä generoitaessa. Tai näin se taitaa paraikaa ns. normiflowssa mennä?

Reply:
Joo, tossa tapauksessa kun tullaan uudella enrollmentilla niin se asetetaan kahdesti. Jos lähtee jo olemassa olevalla enrollmentilla liikkeelle niin sillon status muuttuu (esim. tuo cancel -> expecting_payment)

Reply:
Palautettu takaisin tää kommentin kera